python/ray/data/impl/pipeline_executor.py

```diff
@@ -12,24 +12,31 @@
     from ray.data.dataset_pipeline import DatasetPipeline
 
 
+# Temporarily use an actor here to avoid ownership issues with tasks:
+# https://github.com/ray-project/ray/issues/20554
 @ray.remote(num_cpus=0, placement_group=None)
```

Comment:
Note that @stephanie-wang's PR that ports stage task launching to a threadpool will also fix this by launching all tasks from the driver, and she'll have to resolve some merge conflicts here.

Reply:
This might be an alternative to the thread pools as well, though it's not tolerant of the actor failure.